hibernate-core/src/test/java/org/hibernate/test/legacy/FooBarTest.java

```diff
@@ -615,7 +616,7 @@ public void testQuery() throws Exception {
 		s.save(bar);
 		s.save(baz);
 		list = s.createQuery(
-				" from Bar bar where bar.baz.count=667 and bar.baz.count!=123 and not bar.baz.name='1-E-1'"
+				" from Bar bar where bar.baz.count=667 and bar.baz.count<>123 and not bar.baz.name='1-E-1'"
```

Comment:
TeradataDialect#getForUpdateString returns an empty string. Also TeradataDialect#applyLocksToSql prepends the ForUpdateFragment to the SQL. This overrides the Dialect#applyLocksToSql which appends the ForUpdateFragment to the SQL. 
So, I think its ok as is. At least, it was enough to get me through the test suite. Let me know if you disagree.

Regarding the "<>",  I'm ok with skipping the test.